### PR TITLE
Added CSync class as high level CV wrapper

### PR DESCRIPTION
**Pull request by ethouris**
1. Added `CSync` class. This is predicted to be used exclusively in the local context. It should wrap a conditional variable and the associated mutex in order to perform an operation on them together.

2. The use of `CSync` class replaces every case of using `pthread_cond_*` functions directly (small exception is in the timer code).

### CODE REVIEW

File: srtcore/sync.h [hunk 1]
```diff
@@ -174,6 +174,9 @@ Duration<steady_clock> seconds_from(int64_t t_s);
 
 inline bool is_zero(const TimePoint<steady_clock>& t) { return t.is_zero(); }
 
+timespec us_to_timespec(const uint64_t time_us);
+
+
```
Comment (maxsharabayko) [inline srtcore/sync.h]: Declaration of `us_to_timespec()` is not needed. Only used inside the sync.cpp

File: srtcore/sync.cpp [hunk 2]
```diff
@@ -98,6 +98,115 @@ int64_t get_cpu_frequency()
...
+    bool signaled = SyncEvent::wait_for(m_cond, &m_mutex->ref(), delay) != ETIMEDOUT;
+    THREAD_RESUMED();
+
+    return signaled;
+}
+
+void CSync::lock_signal()
+{
+    // We expect m_nolock == true.
+    lock_signal(*m_cond, *m_mutex);
```
Comment (maxsharabayko) [inline srtcore/sync.cpp]: `m_nolock` is something that will be added in the future? Or in the debug logic?

Comment (ethouris) [reply srtcore/sync.cpp]: Right. The comment remained after I purged this from the thread logging implementation stuff.

File: srtcore/sync.h (hunk 1 above)
Comment (ethouris) [reply srtcore/sync.h]: Right. Can be static.

**Comment by ethouris** [comment]
No. This is not its purpose. The purpose of this class is to serve as a local variable wrapping "temporarily" a condition variable and mutex, usually through CGuard (UniqueLock in future). Max will provide another class, together with new thread libraries (#800) that will allow to combine mutex and cv, this time usable as a field that replace separate fields of mutex and cv. However, this solution will not be applicable in every place - only where the use of mutex+cv is simple. This will also replace the use of current `CSync::lock_signal` provided here in these simple cases.

**Pull request merged**